terraform/README.md

```diff
@@ -1,56 +1,91 @@
 
-3. Configure teleport:
+    _NOTE_: Once PR is merged, we'll replace SHA with v1.0.0
 
-```bash
-tctl create example/teleport.yaml
-tctl auth sign --format=file --user=terraform --out=terraform-identity --ttl=10h
-```
+3. Install [`Terraform`](https://learn.hashicorp.com/tutorials/terraform/install-cli) v1.1.0+. Alternatively, you can use [`tfenv`](https://github.com/tfutils/tfenv). Please note that on Mac M1 you need to specify `TFENV_ARCH` (ex: `TFENV_ARCH=arm64 tfenv install 1.1.6`).
 
-Move generated keys to the desired location.
+4. Clone the plugin:
 
-4. If you desire to use an example for testing:
+    ```bash
+    git clone git@github.com:gravitational/teleport-plugins
+    ```
 
-```bash
-cp example/vars.tfvars.example example/vars.tfvars
-```
+5. Build and install the plugin:
+
+    ```bash
+    cd teleport-plugins/terraform
+    make install
+    ```
 
-Edit `vars.tfvars` and set path to certificate files which were generated in the previous step.
+6. Run tests:
 
-# Regenerating the schema
+    ```bash
+    make test
+    ```
+
+# Updating the provider
+
+Run:
 
 ```
-go install github.com/gravitational/protoc-gen-terraform
-make gen-schema
+make gen-tfschema
 ```
 
-Please note that `ProvisionTokenV2.Allow` field is defined lowercase in `.proto` file (`repeated TokenRule allow = 2 [ (gogoproto.jsontag) = "allow,omitempty" ];`). You need to manually patch `types_tfschema.go`, set `Name: "Allow"` in provision token metadata struct (`GenSchemaMetaProvisionTokenV2()`) until it is fixed.
+This will generate `types_tfschema.go` from a current API `.proto` file, and regenerate the provider code.
 
-# Usage
+# Playing with examples locally
 
-See `example/*.tf` for available configuration options. `make apply` to do an initial application of this configuration to your Terraform cluster.
+1. Start Teleport.
 
-# Testing
+    ```
+    teleport start
+    ```
 
-`TF_ACC=true` is required to switch Terraform to acceptance test mode. Terraform 1.0.0+ needs to be available on the host machine.
+1. Create Terraform user and role:
 
-```
-TF_ACC=true go test test/*
-```
+    ```
+    tctl create example/terraform.yaml
+    tctl auth sign --format=file --user=terraform --out=/tmp/terraform-identity --ttl=10h
+    ```
+
+1. Create `main.tf` file:
+
+    ```
+    cp examples/main.tf.example examlpes/main.tf
+    ```
+
+    Please note that target identity file was exported to `/tmp/terraform-identity` on previous step. If you used another location, please change in in `main.tf`.
+
+1. Create sample resources:
+
+    ```
+    cp examples/user.tf.example examples/user.tf
+    cp examples/role.tf.example examples/role.tf
+    cp examples/provision_token.tf.example examples/provision_token.tf
+    ```
+
+    Please note that some resources require preliminary setup steps.
+
+1. Apply changes:
+
+    ```
+    make apply
+    ```
+
+1. Make changes to .tf files and run:
+
+    ```
+    make reapply
+    ```
```

Comment:
I think a clean up step would be nice here

```suggestion
1. Make changes to .tf files and run:

    ```
    make reapply
    ```

1. Clean up:

    ```
    make destroy
    ```
```